src/components/Onboarding/Screens/Apps/AgreementSettings.js

```diff
@@ -1,19 +1,166 @@
+  Info,
+  Link,
+  TextInput,
+  textStyle,
+  useTheme,
+} from '@1hive/1hive-ui'
 import { useOnboardingState } from '@providers/Onboarding'
 import Navigation from '../../Navigation'
+import { DurationFields, Header } from '../../kit'
 
-function AgreementSettings({ title }) {
-  const { onBack, onNext, step, steps } = useOnboardingState()
+const MAX_TITLE_LENGTH = 50
+const MAX_CONTENT_LENGTH = 8000
+
+const reduceFields = (fields, [field, value]) => {
+  switch (field) {
+    case 'title':
+      return { ...fields, title: value }
+    case 'content':
+      return { ...fields, content: value }
+    case 'challengePeriod':
+      return { ...fields, challengePeriod: value }
+    default:
+      return fields
+  }
+}
+
+const validateAgreementSettings = (title, content, challengePeriod) => {
+  if (!title.trim()) {
+    return 'Please add a title.'
+  }
+  if (!content.trim()) {
+    return 'Please add a content.'
+  }
+  if (!challengePeriod) {
+    return 'Please add a challenge period.'
+  }
+  return null
+}
+
+function AgreementSettings() {
+  const theme = useTheme()
+  const {
+    config,
+    onBack,
+    onNext,
+    onConfigChange,
+    step,
+    steps,
+  } = useOnboardingState()
+  const [formError, setFormError] = useState()
+  const [{ title, content, challengePeriod }, updateField] = useReducer(
+    reduceFields,
+    config.agreement
+  )
+
+  const handleTitleChange = useCallback(
+    ({ target: { value } }) => {
+      if (value.length <= MAX_TITLE_LENGTH) {
+        setFormError(null)
+        updateField(['title', value])
+      }
+    },
+    [updateField]
+  )
+
+  const handleContentChange = useCallback(
+    ({ target: { value } }) => {
+      if (value.length <= MAX_CONTENT_LENGTH) {
+        setFormError(null)
+        updateField(['content', value])
+      }
+    },
+    [updateField]
+  )
+
+  const handleChallengePeriod = useCallback(
+    value => {
+      setFormError(null)
+      updateField(['challengePeriod', value])
+    },
+    [updateField]
+  )
+
+  const handleNextClick = useCallback(() => {
+    const error = validateAgreementSettings(title, content, challengePeriod)
+    setFormError(error)
+
+    if (!error) {
+      onConfigChange('agreement', {
+        title,
+        content,
+        challengePeriod,
+      })
+      onNext()
+    }
+  }, [onConfigChange, challengePeriod, content, onNext, title])
 
   return (
     <div>
-      {title}
+      <Header
+        title="Configure Community Agreement"
+        subtitle="Create the character of your DAO"
+      />
+      <Field label="Title" required>
+        <TextInput value={title} onChange={handleTitleChange} autofocus wide />
+      </Field>
+      <Field label="Content/Covenant" required>
+        <TextInput
+          value={content}
+          placeholder="Define the mission, vision, and values of your community using
+          markdown format."
+          onChange={handleContentChange}
+          multiline
+          wide
+        />
```

Comment:
Should we set an initial height to this input?

So that users have an initial space to start writing

Reply:
We discussed with Fiore that would be best to ask for a markdown file to be uploaded and then have a preview button that shows a modal with the document already formatted as it will be shown on the site.